File: problems/3622/solution.cpp
#include <string>
#include <vector>
using namespace std;

class Solution {
 public:
  bool checkDivisibility(int n) {
    string n_string = to_string(n);
    int sum = 0;
    int product = 1;
    for (const char d : n_string) {
      int digit = d - '0';
      sum += digit;
      product *= digit;
    }
    return n % (sum + product) == 0;
  }
};

File: problems/3622/solution.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        n_string: str = str(n)
        sum, product = 0, 1
        for c in n_string:
            sum += int(c)
            product *= int(c)
        return n % (sum + product) == 0
